### ISSUE TITLE
suggestion: numeric + percantage combos

### ISSUE THREAD

**Issue by red-eye-jedi** [lejeffe/MonsterHP#39]
most bosses having numeric health is ideal, however for certain bosses that have mechanics change at certain percentage thresholds are better off having numeric health off.
could there be a command line you could add to individual npcs to allow them to be either numeric or percentage specifically?

An example being able to type the following into `NPC names`     Corporeal Beast:numeric, Akkha:percentage,

The commands would be super helpful as toggling between numeric and percentage for different activities is tedious, especially when you forget lol

**Comment by xines** [comment]
Thanks for the suggestion. I already thought about this some while ago but was not sure if people even found need in such behavior or if it would complicate the easy of use too much, as of right now to implement this way of being able to dynamically differentiate between numeric and percentage per NPC it would require a little good portion of work.

I had in mind to have it always default to '**percentage**' `unless set to numeric checkbox` like it does currently, and then allow the users to define if they want to display a specific NPC numeric value like:

`monstername:num`
but i was thinking to take this a step further and have it support wildcard matching, like:
`mons*:num`

So no need to specify percentage as it's the default: `monstername1,monstername2` etc.

I'm not sure yet what would look the best either, personally i like `:num` but `:number` or even `:n` would do the job, though the more characters the bigger the list becomes and it might confuse people with bigger lists.

**Comment by red-eye-jedi** [comment]
Wildcard matching seems like a good way forward, you could potentially add
a short line of text to explain how it works.
Npc name or id followed by :n, :num, :numeric, :p, :percent and :percentage.
Other plugins such as better npc highlights implement this in some form for
their respective settings

**Comment by xines** [comment]
Yeah a quick feature explanation would probably help some gamers.

I guess something like checking for the first character to be `n` after colon would do the job, so it wouldn't matter if people simply put gibberish as long as first character is n for numeric. 😄

**Comment by red-eye-jedi** [comment]
That sounds like it would be a good idea lol, you're doing God's work. 👌 🫡

**Comment by xines** [comment]
Just a quick update on this.

I made this already, but need to test it more to make sure it works the way it's supposed to. 😄

**Comment by red-eye-jedi** [comment]
Wow, what a beast. Thanks for taking the time to add my suggestion. Very
much looking forward to this. Great addition and great plugin sir!

**Comment by xines** [comment]
> Wow, what a beast. Thanks for taking the time to add my suggestion. Very much looking forward to this. Great addition and great plugin sir!

Thanks i appreciate the kind words, though this plugin was not created by me but by Le Jeffe, i'm simply just a collaborator trying to maintain it and add some quality of life / improvements.

I am just waiting on @lejeffe to push the latest update to the plugin hub we should be gucci 🔥 👍

**Comment by red-eye-jedi** [comment]
Appreciate it all the same! 2 legends (;
Looking forward to it. Thanks again <3

**Comment by xines** [comment]
> Appreciate it all the same! 2 legends (; Looking forward to it. Thanks again <3

Should be active on Runelite now, enjoy king. 🦾
